src/auth/hooks/mutations/useSignIn.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState } from "react";
+import { useAuthContext } from "auth/context/auth.hook";
+import { getCredentials } from "../../lib";
+
+interface Options {
+  onError?: (msg: string) => void;
+  onSuccess?: () => void;
+}
+
+export function useSignIn({ onError, onSuccess }: Options = {}) {
+  const { setSessionUser } = useAuthContext();
+  const [loading, setLoading] = useState(false);
+  const signIn = useCallback((email: string, password: string) => {
+    setLoading(true);
+    getCredentials({ email, password })
+      .then((data) => {
+        if (data) {
+          setSessionUser(data);
+
+          if (onSuccess) {
+            onSuccess();
+          }
+        }
+      })
+      .catch((e) => {
+        if (onError) {
+          onError(e.message);
+        }
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
```

Comment:
If it works, let's leave this. But I'm surprised this dependency list is empty, I'd have thought it needs to include everything referenced within the callback. Do we not have the react linter enabled?